Lib/zipfile/__init__.py

```diff
@@ -1772,9 +1772,6 @@ def _extract_member(self, member, targetpath, pwd):
             # filter illegal characters on Windows
             arcname = self._sanitize_windows_name(arcname, os.path.sep)
 
-        if not arcname:
```

Comment:
I suspect we should keep this but do something like:

`if not arcname and member.is_dir():`

instead.

We shouldn't allow a name that normalizes to empty be treated as a file, the code below could otherwise be convinced to create a _file_ called targetpath with the empty name member's contents in that situation, which is not right.